### ISSUE TITLE
PSReadLine gets disabled when using Intel UHD Graphics driver v31.0.101.5448

### ISSUE THREAD

**Issue by sba923** [PowerShell/PSReadLine#4015]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Run PowerShell 7.4.2 (or 7.5.0-preview.2) within Windows Terminal Preview v1.21.1272.0 with Intel UHD Graphics driver v31.0.101.5448 dd. 2024-05-03.

Upon startup, PowerShell reports a screen reader is present and disables PSReadLine.

This problem was not present with v31.0.101.5445 dd. 2024-04-20 of the Intel driver.

### Expected behavior

```console
PowerShell loads PSReadLine normally.
```

### Actual behavior

```console
Warning: PowerShell detected that you might be using a screen reader and has disabled PSReadLine for compatibility purposes. If you want to re—enable it, run 'Import—module PSReadLine'.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.2
PSEdition                      Core
GitCommitId                    7.4.2
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by SteveL-MSFT** [comment]
If the Intel graphics driver registers itself as a screen reader, then (unfortunately), this is the current behavior as PSReadLine has challenges with screen readers due to the frequency of it redrawing

**Comment by daxian-dbw** [comment]
The check on screen reader is in PowerShell code, [here](https://github.com/PowerShell/PowerShell/blob/e05542b1a2445f61dfb7390f98d9e12ce89ab979/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs#L1626-L1648). It doesn't sound right the intel graphics drive would register itself as a screen reader, but something does if that check returns true, and it's by design for PowerShell to not load PSReadLine in that case.

**Comment by sba923** [comment]
> If the Intel graphics driver registers itself as a screen reader, then (unfortunately), this is the current behavior as PSReadLine has challenges with screen readers due to the frequency of it redrawing

Point taken; thanks for the justification (TBH I was wondering what the coupling could be between PSReadLine and screen readers...).

It was also my plan to report the issue to Intel (haven't done it yet).

For now, I've reverted to the previous version of the Intel driver.

**Comment by sba923** [comment]
> If the Intel graphics driver registers itself as a screen reader, then (unfortunately), this is the current behavior as PSReadLine has challenges with screen readers due to the frequency of it redrawing

Point taken; thanks for the justification (TBH I was wondering what the coupling could be between PSReadLine and screen readers...).

It was also my plan to report the issue to Intel (haven't done it yet).

For now I've reverted to the previous version of the Intel driver.

**Comment by sba923** [comment]
FYI problem is solved in version 31.0.101.5518 of the Intel driver (which isn't found by Intel DSA -- yet? -- that update was suggested by Intel support).

**Comment by daxian-dbw** [comment]
@sba923 Thanks for reporting back on this issue!
